src/services/admin.js

```diff
@@ -1539,4 +1608,391 @@ module.exports = class AdminService {
+				}
+			} else if (orgId && orgId !== 'all') {
+				// Clear all cache for specific org
+				patterns.push(`tenant:${tenantCode}:org:${orgId}:*`)
+			} else {
+				// Clear all cache for tenant
+				patterns.push(`tenant:${tenantCode}:*`)
+			}
+
+			// Execute cache clearing
+			for (const pattern of patterns) {
+				const keys = await this.scanKeys(redisClient, pattern)
+				if (keys.length > 0) {
+					await redisClient.del(...keys)
+					clearedKeys += keys.length
+				}
+			}
+
+			console.log(`💾 Admin cache clear: ${clearedKeys} keys cleared for patterns: ${patterns.join(', ')}`)
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'CACHE_CLEARED_SUCCESSFULLY',
+				result: {
+					clearedKeys,
+					patterns,
+					tenantCode,
+					orgId,
+					namespace,
+					timestamp: new Date().toISOString(),
+				},
+			})
+		} catch (error) {
+			console.error('Error clearing cache:', error)
+			return responses.failureResponse({
+				statusCode: httpStatusCode.internal_server_error,
+				message: 'CACHE_CLEAR_FAILED',
+				responseCode: 'SERVER_ERROR',
+				result: { error: error.message },
+			})
+		}
+	}
+
+	/**
+	 * Warm up cache for tenant/organization
+	 * @method
+	 * @name warmUpCache
+	 * @param {Object} options - Warm up options
+	 * @returns {JSON} - Cache warm up response
+	 */
+	static async warmUpCache({ tenantCode, orgId, adminTenantCode, adminOrgId }) {
+		try {
+			const warmupResults = {
+				entityTypes: 0,
+				organizations: 0,
+				platformConfig: 0,
+				permissions: 0,
+				forms: 0,
+			}
+
+			// Warm up EntityTypes cache
+			try {
+				const entityTypeHelper = require('@helpers/entityTypeCache')
+				await entityTypeHelper.warmUpEntityTypesCache(tenantCode, orgId)
+				warmupResults.entityTypes = 1
+			} catch (error) {
+				console.error('EntityTypes cache warmup failed:', error)
+			}
+
+			// Warm up Organizations cache
+			try {
+				const orgQueries = require('@database/queries/organisationExtension')
+				const orgData = await orgQueries.findOne({ organization_id: orgId }, tenantCode)
+				if (orgData) {
+					await cacheHelper.organizations.set(tenantCode, orgId, orgId, orgData)
+					warmupResults.organizations = 1
+				}
+			} catch (error) {
+				console.error('Organizations cache warmup failed:', error)
+			}
+
+			// Warm up Platform Config cache
+			try {
+				const formQueries = require('@database/queries/form')
+				const configData = await formQueries.findOneForm({ code: 'platformConfig' }, tenantCode)
+				if (configData) {
+					await cacheHelper.platformConfig.set(tenantCode, orgId, configData)
+					warmupResults.platformConfig = 1
+				}
+			} catch (error) {
+				console.error('Platform Config cache warmup failed:', error)
+			}
+
+			// Warm up Permissions cache
+			try {
+				const permissionsQueries = require('@database/queries/permissions')
+				const roles = ['mentee', 'mentor', 'org_admin', 'session_manager']
```

Comment:
@sumanvpacewisdom dont hard code the roles here, get from the request body, or set for all the roles